pkg/kubelet/volumemanager/reconciler/reconciler.go

```diff
@@ -604,7 +604,9 @@ func getDeviceMountPath(volume *reconstructedVolume) (string, error) {
 
 func (rc *reconciler) updateStates(volumesNeedUpdate map[v1.UniqueVolumeName]*reconstructedVolume) error {
 	// Get the node status to retrieve volume device path information.
-	rc.updateDevicePath(volumesNeedUpdate)
+	if rc.kubeClient != nil {
+		rc.updateDevicePath(volumesNeedUpdate)
+	}
```

Comment:
I am not 100% sure if it is entirely safe to skip updating node objects silently and still do rest of reconstruction. We need to rethink about what volume-manager features are supposed to be working in standalone mode. Please see my comment - https://github.com/kubernetes/kubernetes/issues/108063#issuecomment-1063936254

Reply:
@gnufied
Does this statement
`If Kubelet is running in standalone mode, there is no apiserver and kubeClient should be nil.` of [this](https://github.com/kubernetes/kubernetes/issues/108063#issuecomment-1056064383) comment give the answer of  your question about **what volume-manager features are supposed to be working in standalone mode ?**

Reply:
It does not. I am waiting for some clarification from sig-node about, which volume features are available in standalone mode. Without understanding that - crashing the kubelet is better than silently incorrectly accounting volumes that are in-use by node.